tests/apps/transaction_nonActor_apis/app.go

```diff
@@ -0,0 +1,68 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+)
+
+const appPort = 3000
+
+const stateURL = "http://localhost:3500/v1.0/state/statestore"
+const transactionURL = "http://localhost:3500/v1.0/state/statestore/transaction"
+
+type testCommandRequest struct {
+	Message string `json:"message,omitempty"`
+}
+
+type appResponse struct {
+	Message   string `json:"message,omitempty"`
+	StartTime int    `json:"start_time,omitempty"`
+	EndTime   int    `json:"end_time,omitempty"`
+}
+
+// indexHandler is the handler for root path
+func indexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println("indexHandler is called")
+
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(appResponse{Message: "OK"})
+}
+
+// testHandler is the handler for end-to-end test entry point
+// test driver code call this endpoint to trigger the test
+func testHandler(w http.ResponseWriter, r *http.Request) {
+	log.Printf("Processing transaction for %s", r.URL.RequestURI())
+}
+
+// epoch returns the current unix epoch timestamp
+func epoch() int {
+	return (int)(time.Now().UTC().UnixNano() / 1000000)
+}
+
+// appRouter initializes restful api router
+func appRouter() *mux.Router {
+	router := mux.NewRouter().StrictSlash(true)
+
+	router.HandleFunc("/", indexHandler).Methods("GET")
+	router.HandleFunc("/tests/transasction", testHandler).Methods("POST")
```

Comment:
Can we reuse stateapp implementation - https://github.com/dapr/dapr/tree/master/tests/apps/stateapp ?
Please add this endpoint to this app.

Reply:
> Thanks for your contribution! We also need to add e2e test for this - https://github.com/dapr/dapr/blob/master/tests/docs/writing-e2e-test.md

Is there an implementation or test file that I can add to for the gRPC test? I think the http test is ready.

Reply:
yes there is. you can find it in the other comment - https://github.com/dapr/dapr/blob/master/tests/e2e/stateapp/stateapp_test.go